lib/host/wasi/wasifunc.cpp

```diff
@@ -1846,5 +1846,139 @@ Expect<uint32_t> WasiSockShutdown::body(Runtime::Instance::MemoryInstance *,
   return __WASI_ERRNO_SUCCESS;
 }
 
+Expect<uint32_t>
+WasiGetAddrinfo::body(Runtime::Instance::MemoryInstance *MemInst,
+                      uint32_t NodePtr, uint32_t NodeLen, uint32_t ServicePtr,
+                      uint32_t ServiceLen, uint32_t HintsPtr, uint32_t ResPtr,
+                      uint32_t MaxResLength, uint32_t ResLengthPtr) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  const char *Node = nullptr;
+  if (NodeLen != 0) {
+    Node = MemInst->getPointer<const char *>(NodePtr, NodeLen);
+  }
+  const char *Service = nullptr;
+  if (ServiceLen != 0) {
+    Service = MemInst->getPointer<const char *>(ServicePtr, ServiceLen);
+  }
+  // service and node can not be nullptr at the same time
+  if (Service == nullptr && Node == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  auto *const Hint = MemInst->getPointer<const struct __wasi_addrinfo_t *>(
+      HintsPtr, sizeof(struct __wasi_addrinfo_t));
+  auto *const ResLength = MemInst->getPointer<__wasi_size_t *>(
+      ResLengthPtr, sizeof(__wasi_size_t *));
+
+  struct addrinfo *SysAddrinfoRes = nullptr;
+  if (auto Res =
+          Env.getAddrInfo(Node, Service, Hint, &SysAddrinfoRes, ResLength);
+      unlikely(!Res)) {
+    return Res.error();
+  }
+  if (*ResLength > MaxResLength) {
```

Comment:
Looks like your `ResLength` is used from this line. We can calculate `ResLength` in this section.